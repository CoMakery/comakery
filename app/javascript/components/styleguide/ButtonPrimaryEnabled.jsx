import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import Button from './Button'

class ButtonPrimaryEnabled extends React.Component {
  render() {
    const {className, ...other} = this.props

    const classnames = classNames(
      'button__primary__enabled',
      className
    )

    return (
      <React.Fragment>
        <Button className={classnames} {...other} />
      </React.Fragment>
    )
  }
}

ButtonPrimaryEnabled.propTypes = {
  className: PropTypes.string
}
ButtonPrimaryEnabled.defaultProps = {
  className: ''
}
export default ButtonPrimaryEnabled
