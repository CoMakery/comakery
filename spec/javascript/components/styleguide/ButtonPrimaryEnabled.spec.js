import React from 'react'
import { shallow, mount, render } from 'enzyme'
import ButtonPrimaryEnabled from 'components/styleguide/ButtonPrimaryEnabled'

describe('ButtonPrimaryEnabled', () => {
  it('renders correctly without props', () => {
    const wrapper = shallow(<ButtonPrimaryEnabled />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.exists('.button__primary__enabled')).toBe(true)
  })

  it('renders correctly with custom className', () => {
    const wrapper = shallow(<ButtonPrimaryEnabled className="__test" />)

    expect(wrapper.exists('.button__primary__enabled.__test')).toBe(true)
  })
})
