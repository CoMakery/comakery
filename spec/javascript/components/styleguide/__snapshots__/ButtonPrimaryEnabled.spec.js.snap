// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPrimaryEnabled renders correctly without props 1`] = `
<Fragment>
  <Button
    className="button__primary__enabled"
    disabled={false}
    type="button"
    value="publish"
  />
</Fragment>
`;
